jobs/ci-kubernetes-bazel-build.sh

```diff
@@ -34,7 +34,14 @@ if [[ "${rc}" == 0 ]]; then
     echo "Kubernetes version missing; not uploading ci artifacts."
     rc=1
   else
-    bazel run //:ci-artifacts -- "gs://kubernetes-release-dev/bazel/${version}" && rc=$? || rc=$?
+    push_build="../release/push-build.sh"
```

Comment:
I'm a little confused. Does this actually work at CI time? Even given your changes in https://github.com/kubernetes/test-infra/pull/2731, I'm not sure what mechanism pulls the `release` repository. Is that done automatically for all jobs? I thought that this job only pulled `k8s.io/kubernetes`, so this looks like something that would only work locally.